code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm

```diff
@@ -147,36 +147,55 @@ Doesn't work on other aliens/AI.*/
 	desc = "Drench an object in acid, destroying it over time."
 	plasma_cost = 200
 	action_icon_state = "alien_acid"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/acid.dmi'
+	active = FALSE
 
 /obj/effect/proc_holder/alien/acid/on_gain(mob/living/carbon/user)
 	add_verb(user, /mob/living/carbon/proc/corrosive_acid)
 
 /obj/effect/proc_holder/alien/acid/on_lose(mob/living/carbon/user)
 	remove_verb(user, /mob/living/carbon/proc/corrosive_acid)
+	remove_ranged_ability()
 
-/obj/effect/proc_holder/alien/acid/proc/corrode(atom/target,mob/living/carbon/user = usr)
-	if(target in oview(1,user))
-		if(target.acid_act(200, 100))
-			user.visible_message(span_alertalien("[user] vomits globs of vile stuff all over [target]. It begins to sizzle and melt under the bubbling mess of acid!"))
-			return 1
-		else
-			to_chat(user, span_noticealien("You cannot dissolve this object."))
-
-
-			return 0
+/obj/effect/proc_holder/alien/acid/fire(mob/living/user)	
+	if(active)
+		user.balloon_alert(user, "acid glands relaxed")
+		remove_ranged_ability()
 	else
-		to_chat(src, span_noticealien("[target] is too far away."))
-		return 0
+		user.balloon_alert(user, "acid glands ready")
+		add_ranged_ability(user)
 
+/obj/effect/proc_holder/alien/acid/proc/check_target(atom/target)
+	return isturf(target) || isobj(target)
 
-/obj/effect/proc_holder/alien/acid/fire(mob/living/carbon/alien/user)
-	var/O = input("Select what to dissolve:","Dissolve",null) as obj|turf in oview(1,user)
-	if(!O || user.incapacitated())
-		return 0
-	else
-		return corrode(O,user)
+/obj/effect/proc_holder/alien/acid/InterceptClickOn(mob/living/caller, params, atom/target)
+	if(..())
+		return TRUE
+	if(!iscarbon(ranged_ability_user) || ranged_ability_user.stat != CONSCIOUS)
+		remove_ranged_ability()
+		return TRUE
+	var/mob/living/carbon/user = ranged_ability_user
+	if(!target)
+		return TRUE
+	if(!check_target(target))
+		user.balloon_alert(user, "can't acid this!")
+		return TRUE
+	if(get_dist(user, target) > 1) //close?
+		user.balloon_alert(user, "too far!")
+		return TRUE	
+	if(user.getPlasma() < plasma_cost) //pay up
+		user.balloon_alert(user, "not enough plasma!")
+		remove_ranged_ability()
+		return TRUE
+	user.adjustPlasma(-plasma_cost)
+	if(target.acid_act(200, 100)) //do our thing
+		user.visible_message(span_alertalien("[user] vomits globs of vile stuff all over [target]. It begins to sizzle and melt under the bubbling mess of acid!"))
+		remove_ranged_ability()
+		return FALSE
+	user.balloon_alert(user, "cannot disolve") //if we can't do our thing just don't
+	return TRUE
```

Comment:
```suggestion
/obj/effect/proc_holder/alien/acid/InterceptClickOn(mob/living/caller, params, atom/target)
	if(..())
		return TRUE

	// Oh great, another unconscious alien. Let's just remove the ranged ability.
	if(!iscarbon(ranged_ability_user) || ranged_ability_user.stat != CONSCIOUS)
		remove_ranged_ability()
		return TRUE

	var/mob/living/carbon/user = ranged_ability_user

	// If there's no target, why even bother?
	if(!target || !check_target(target))
		user.balloon_alert(user, "can't acid this!")
		return TRUE

	// Too far away
	if(get_dist(user, target) > 1)
		user.balloon_alert(user, "too far!")
		return TRUE

	// Plasma cost check. Because aliens need a currency too, apparently.
	if(user.getPlasma() < plasma_cost)
		user.balloon_alert(user, "not enough plasma!")
		remove_ranged_ability()
		return TRUE

	// Let's just get this over with. Apply the acid effect and move on.
	user.adjustPlasma(-plasma_cost)
	if(target.acid_act(200, 100))
		user.visible_message(span_alertalien("[user] vomits globs of vile stuff all over [target]. It begins to sizzle and melt under the bubbling mess of acid!"))
		remove_ranged_ability()
		return FALSE

	// Fantastic. It didn't even work.
	user.balloon_alert(user, "cannot disolve")
	return TRUE
```
Why: Looked like a mess + had too much nesting